lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -794,37 +785,48 @@ protected function getRenameIndexSQL($oldIndexName, Index $index, $tableName)
      *
      * @link http://msdn.microsoft.com/en-us/library/ms180047%28v=sql.90%29.aspx
      *
      * @param string      $name       The name of the property to add.
-     * @param string|null $value      The value of the property to add.
-     * @param string|null $level0Type The type of the object at level 0 the property belongs to.
-     * @param string|null $level0Name The name of the object at level 0 the property belongs to.
-     * @param string|null $level1Type The type of the object at level 1 the property belongs to.
-     * @param string|null $level1Name The name of the object at level 1 the property belongs to.
+     * @param string      $value      The value of the property to add.
+     * @param string      $level0Type The type of the object at level 0 the property belongs to.
+     * @param string      $level0Name The name of the object at level 0 the property belongs to.
+     * @param string      $level1Type The type of the object at level 1 the property belongs to.
+     * @param string      $level1Name The name of the object at level 1 the property belongs to.
      * @param string|null $level2Type The type of the object at level 2 the property belongs to.
      * @param string|null $level2Name The name of the object at level 2 the property belongs to.
      *
      * @return string
      */
     public function getAddExtendedPropertySQL(
         $name,
-        $value = null,
-        $level0Type = null,
-        $level0Name = null,
-        $level1Type = null,
-        $level1Name = null,
-        $level2Type = null,
-        $level2Name = null
+        $value,
+        string $level0Type,
+        string $level0Name,
+        string $level1Type,
+        string $level1Name,
+        ?string $level2Type = null,
+        ?string $level2Name = null
     ) {
-        return 'EXEC sp_addextendedproperty ' .
-            'N' . $this->quoteStringLiteral($name) . ', N' . $this->quoteStringLiteral((string) $value) . ', ' .
-            'N' . $this->quoteStringLiteral((string) $level0Type) . ', ' . $level0Name . ', ' .
-            'N' . $this->quoteStringLiteral((string) $level1Type) . ', ' . $level1Name . ', ' .
-            'N' . $this->quoteStringLiteral((string) $level2Type) . ', ' . $level2Name;
+        return 'EXEC sp_addextendedproperty'
+            . ' N' . $this->quoteStringLiteral($name) . ', N' . $this->quoteStringLiteral((string) $value)
+            . ', N' . $this->quoteStringLiteral($level0Type)
+            . ', ' . $this->quoteSingleIdentifierAsStringLiteral($level0Name)
+            . ', N' . $this->quoteStringLiteral($level1Type)
+            . ', ' . $this->quoteSingleIdentifierAsStringLiteral($level1Name)
+            . ($level2Type !== null || $level2Name !== null
+                ? ', N' . $this->quoteStringLiteral((string) $level2Type)
+                  . ', ' . $this->quoteSingleIdentifierAsStringLiteral((string) $level2Name)
```

Comment:
The only line related to the fix are the ones with `quoteSingleIdentifierAsStringLiteral`, right?

Reply:
I think `getCommentOnTableSQL` at least as well